src/ui/model_viewer_widget.cc

```diff
@@ -71,12 +71,9 @@ inline Eigen::Vector4f IndexToRGB(const size_t index) {
 void BuildImageModel(const Image& image, const Camera& camera,
                      const float image_size, const Eigen::Vector4f& plane_color,
                      const Eigen::Vector4f& frame_color,
-                     TrianglePainter::Data* triangle1,
-                     TrianglePainter::Data* triangle2, LinePainter::Data* line1,
-                     LinePainter::Data* line2, LinePainter::Data* line3,
-                     LinePainter::Data* line4, LinePainter::Data* line5,
-                     LinePainter::Data* line6, LinePainter::Data* line7,
-                     LinePainter::Data* line8) {
+                     std::vector<TrianglePainter::Data>& triangle_data,
```

Comment:
output arguments should be pointers (Google C++ coding guideline).

Reply:
I'm not too familiar with the Google C++ coding guidelines, but it seems like [this section here](https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs) recommends references. Am I checking in the wrong place?

Or do you mean that I should consider line_data to be an optional output parameter and then remove the show_lines parameter and instead pass nullptr as line_data when not required?

Reply:
Apparently they updated their guidelines and I am pretty sure the old convention was to use pointers. In any case, it's the convention across the entire COLMAP code base, so we should follow it here as well. Thanks.

Reply:
Okay makes sense. In this case it's probably best to add null-checks and to remove the show_lines parameter. Same here, I'll fix this as soon as I get home.